lib/ddtrace/profiling/ext/cpu.rb

```diff
@@ -20,6 +17,28 @@ def self.apply!
           require 'ddtrace/profiling/ext/cthread'
           ::Thread.send(:prepend, Profiling::Ext::CThread)
         end
+
+        def self.unsupported_reason
+          # Note: Only the first matching reason is returned, so try to keep a nice order on reasons -- e.g. tell users
+          # first that they can't use this on macOS before telling them that they have the wrong ffi version
+
+          if RUBY_ENGINE == 'jruby'
+            'JRuby is not supported'
+          elsif RUBY_PLATFORM =~ /darwin/
+            'Feature requires Linux; macOS is not supported'
+          elsif RUBY_PLATFORM =~ /(mswin|mingw)/
+            'Feature requires Linux; Windows is not supported'
+          elsif RUBY_PLATFORM !~ /linux/
+            "Feature requires Linux; #{RUBY_PLATFORM} is not supported"
```

Comment:
Are the `/darwin/` and `/(mswin|mingw)/` branches required, given we have the `/linux/` one, which reports the unsupported platform in its message?

Reply:
Technically no, they aren't needed, but I wanted to be a bit more user-friendly for the most well-known other OSs. I claim that `macOS is not supported` is a bit more user-friendly than `x86_64-darwin19 is not supported` :)